src/t8_forest/t8_forest_adapt.cxx

```diff
@@ -248,26 +254,40 @@ t8_forest_adapt (t8_forest_t forest)
      * array which could be coarsened recursively. */
     el_coarsen = 0;
     /* TODO: this will generate problems with pyramidal elements */
```

Comment:
Is this TODO fixed now?

Reply:
I think so. We are able to coarsen pyramids without any problems.